### ISSUE TITLE
[QNN ISSUE] Error Occurred When Running Llama-3.2-1B-Instruct Model with llama_main (QNN 8A8W Quantization) on Device

### ISSUE THREAD

**Issue by crinex** [pytorch/executorch#6546]
### 🐛 Describe the bug

Dear @shewu-quic 

I followed your [instructions](https://github.com/pytorch/executorch/blob/main/examples/demo-apps/android/LlamaDemo/docs/delegates/qualcomm_README.md) to convert the Llama-3.2-1B-Instruct model to a **QNN backend** **.pte** file. However, during the **On Device** process, I encountered an error as shown in Figure 1 below, and I would like you to help me check this issue.

The command below is what I used to generate the .pte file.

`MODEL_DIR="Llama-3.2-1B-Instruct/original"`
`python -m examples.models.llama.export_llama --checkpoint "${MODEL_DIR}/consolidated.00.pth" -p "${MODEL_DIR}/params.json" -kv --disable_dynamic_shape --qnn --pt2e_quantize qnn_8a8w -d fp32 --metadata '{"get_bos_id":128000, "get_eos_ids":[128009, 128001]}' --num_sharding 8 --output_name="llama3_2_qnn_8a8w.pte"`

Then, I ran the generated file on the device using the command below.

`./llama_main_qnn --model_path /data/local/tmp/llama/llama3_2_qnn_8a8w.pte --tokenizer_path /data/local/tmp/llama/tokenizer.model --prompt "<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n\nYou are a funny chatbot.<|eot_id|><|start_header_id|>user<|end_header_id|>\n\nWhat is the capital of Korea?<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n\n" --seq_len 512 —temperature 0`

Then, an error occurred as shown in Figure 1 below.

![image](https://github.com/user-attachments/assets/de1a51ca-5c87-47ba-80f3-07278023544a)
[Figure 1. The error occurring when running with llama_main (QNN version).]

I would appreciate it if you could help me identify the issue.
Thank you

### Versions

PyTorch version: 2.6.0.dev20241019+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.30.5
Libc version: glibc-2.35

Python version: 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.8.0-40-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        46 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               24
On-line CPU(s) list:                  0-23
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Core(TM) i9-10920X CPU @ 3.50GHz
CPU family:                           6
Model:                                85
Thread(s) per core:                   2
Core(s) per socket:                   12
Socket(s):                            1
Stepping:                             7
CPU max MHz:                          4800.0000
CPU min MHz:                          1200.0000
BogoMIPS:                             6999.82
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512_vnni md_clear flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            384 KiB (12 instances)
L1i cache:                            384 KiB (12 instances)
L2 cache:                             12 MiB (12 instances)
L3 cache:                             19.3 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-23
Vulnerability Gather data sampling:   Mitigation; Microcode
Vulnerability Itlb multihit:          KVM: Mitigation: VMX disabled
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Mitigation; Enhanced IBRS
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Mitigation; TSX disabled

Versions of relevant libraries:
[pip3] executorch==0.5.0a0+16b633b
[pip3] numpy==1.26.4
[pip3] torch==2.6.0.dev20241019+cpu
[pip3] torchao==0.5.0+git0916b5b2
[pip3] torchaudio==2.5.0.dev20241019+cpu
[pip3] torchsr==1.0.4
[pip3] torchvision==0.20.0.dev20241019+cpu
[conda] Could not collect

**Comment by shewu-quic** [comment]
Hi @crinex 
Thanks for reaching out.
I think num_sharding is not necessary for llama3.2 1B.
Could you try to export without using num_sharding?

**Comment by crinex** [comment]
Dear @shewu-quic 

I followed your instructions, and it worked successfully—thank you very much! 
However, the inference results seem unusual. Do you happen to know why that might be?
![image](https://github.com/user-attachments/assets/7ec4bd95-e7c6-43a2-ab00-50bf3ddea650)

Thank you

**Comment by shewu-quic** [comment]
Yes, it is a known issue for llama 3.2. We are also trying to fix it [with QAT](https://github.com/pytorch/executorch/pull/6513). 
If there are any updates, we will let you know as soon as possible.

**Comment by crinex** [comment]
I really appreciate all your efforts. We will also try this method on our end. 
I think it’s fine to close this issue now. 
Thank you!
